### PR TITLE
Line terminator override option

### PR DESCRIPTION
**Pull request by mhollier**

### CODE REVIEW

File: Src/KBCsv.PerformanceTests/KBCsv.PerformanceTests.csproj [hunk 1]
```diff
@@ -14,11 +14,16 @@
...
     <PackageReference Include="xunit" Version="2.3.0-beta1-build3642" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\KBCsv\KBCsv.csproj" />
     <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.0-beta1-build3642" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (kentcb) [inline Src/KBCsv.PerformanceTests/KBCsv.PerformanceTests.csproj]: Any ideas what caused this `ItemGroup` to be added? Seems like it's related to VSTE? Are you able to run tests without it? Bit strange that it was added only to this project...

File: Src/KBCsv/CsvReader.cs [hunk 2]
```diff
@@ -268,6 +268,30 @@ public char ValueSeparator
         }
 
         /// <summary>
+        /// Gets or sets the character used to override the default line terminators (CR/LF).
+        /// <remarks>
+        /// <para>
+        /// If set to NULL, then the default line terminator (CR/LF) is used.
```
Comment (kentcb) [inline Src/KBCsv/CsvReader.cs]: Please change "NULL" to `<see langword="null"/>`.

File: Src/KBCsv/Internal/CsvParser_Async.cs [hunk 3]
```diff
@@ -159,7 +163,13 @@ internal sealed partial class CsvParser
                                 buffer[i] = this.values.GetDataRecordAndClear(headerRecord, this.valueBuilder.GetValueAndClear());
                                 break;
                             }
-                            else if (ch == Constants.LF)
+                            else if (this.lineTerminatorOverride == null && ch == Constants.LF)
```
Comment (kentcb) [inline Src/KBCsv/Internal/CsvParser_Async.cs]: Couldn't these two cases be collapsed into one? That is, `if (ch == Constants.LF || ch == this.lineTerminatorOverride)`

File: Src/KBCsv/CsvReader.cs [hunk 4]
```diff
@@ -268,6 +268,30 @@ public char ValueSeparator
...
+        /// </para>
+        /// </remarks>
+        /// </summary>
+        public char? LineTerminatorOverride
```
Comment (kentcb) [inline Src/KBCsv/CsvReader.cs]: I don't love the naming here. How about just calling it `NewLine` to match the equivalent property in `CsvWriter`. The docs should then make it clear that leaving as `null` will default to treating CR, LF, and CRLF as line endings.

**Comment by mhollier** [comment]
Hi Kent,
Thanks for evaluating this as a possible solution. Since I submitted this, I have since learned there were additional requirements such as multi-character delimiters. As a result, we have decided to implement our own parser to handle these requirements.
Best Regards,
Mark

**Comment by kentcb** [comment]
Thanks for the update Mark.

**Pull request closed without merging**